Use cookie parser to read cookies for matching

RealOrangeOne opened the pull request:
Previously this would match cookies whose values _contained_ the private cookies, rather than simply checking for their existence.

Implementation copied from https://developers.cloudflare.com/workers/examples/extract-cookie-value

RealOrangeOne committed:
Use cookie parser to read cookies for matching

Previously this would match cookies whose values _contained_ the private cookies, rather than simply checking for their existance

tomusher commented:
Nitpick: We're parsing the entire cookie header once for every cookie in the list here - it's not going to be an issue with 2 cookies, and Cloudflare's going to be fast anyway, but maybe worth just iterating through the cookie string once?

RealOrangeOne committed:
Just match cookie names

This makes the parser simpler, and hopefully faster. It also only parses once per request.

tomusher commented:
Nice! LGTM!

balazs-endresz commented on common-caching.js:
```diff
@@ -65,14 +65,9 @@ function requestIsCachable(request) {
 }
 
 function hasPrivateCookie(request) {
+  const requestCookieNames = request.headers.get("Cookie").split(";").map(cookie => cookie.split("=")[0].trim());
```
Wouldn't this fail if `request.headers.get("Cookie")` is null?
